core/src/main/java/org/testcontainers/DockerClientFactory.java

```diff
@@ -210,6 +212,8 @@ public void close() {
         Version version = client.versionCmd().exec();
         log.debug("Docker version: {}", version.getRawValues());
         activeApiVersion = version.getApiVersion();
+        String osType = dockerInfo.getOsType();
+        runningWindowsContainers = StringUtils.isNotBlank(osType) && osType.equals("windows");
```

Comment:
Is this really enough to identify WCOW? Meaning with Docker Desktop on Windows with WSL2 backend, it would return `linux` here?

Reply:
AFAIK, yes, but I'll check later to ensure we have correct assumptions here.

Reply:
Here are both variants, using Docker Desktop on Windows 11, WSL2 enabled (not Hyper-V):

```shell
λ "c:\Program Files\Docker\Docker\DockerCli.exe" -SwitchLinuxEngine
λ docker info --format "{{json .OSType }}"
"linux"
λ "c:\Program Files\Docker\Docker\DockerCli.exe" -SwitchWindowsEngine
λ docker info --format "{{json .OSType }}"
"windows"
```